### ISSUE TITLE
 Pull and parse parent devfile from Kubernetes CRD in library

### ISSUE THREAD

**Issue by yangcao77** [devfile/api#355]
Currently devfile library can only parse parent devfile if uri is specified.

This issue is to support Kubernetes CRD defined under parent.
(Question: should be with type `DevWorkspaceTemplate` or `DevWorkspace` ??)

Needs to get KubeConfig and creates Kubernetes client, and access the Kubernetes CRD from that client.

The parent devfile should be parsed & override & merge into the flattened devfile.

**Comment by maysunfaisal** [comment]
Before starting we should check with the devworkspace operator repo and if they have logic in place for this issue already and needs consolidation..
